File: src/ISUCorp.Core/Kernel/BaseEntity.cs
using System;

namespace ISUCorp.Core.Kernel
{
    public abstract class BaseEntity
    {
        /// <summary>
        /// Entity identifier.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Time when the entity was added at.
        /// </summary>
        public DateTime AddedAt { get; set; }

        /// <summary>
        /// Time when the entity was modified at.
        /// </summary>
        public DateTime? ModifiedAt { get; set; }
    }
}

File: src/ISUCorpApp/src/app/models/base.entity.ts
export class BaseEntity {
    id?: number;
    addedAt: Date;
    modifiedAt: Date;

    constructor(addedAt: Date, modifiedAt: Date, id: number = null){
        this.id = id;
        this.addedAt = addedAt;
        this.modifiedAt = modifiedAt;
    }
}
